plugins/out_s3/s3.c

```diff
@@ -1186,6 +1227,18 @@ static int put_all_chunks(struct flb_s3 *ctx)
                 return -1;
             }
 
+            if (ctx->compression != FLB_AWS_COMPRESS_NONE) {
+                /* Map payload */
+                ret = flb_aws_compression_compress(ctx->compression, buffer, buffer_size, &payload_buf, &payload_size);
+                if (ret == -1) {
+                    flb_plg_error(ctx->ins, "Failed to compress data, uploading uncompressed data instead to prevent data loss");
+                } else {
+                    flb_plg_info(ctx->ins, "Pre-compression upload_chunk_size= %d, After compression, chunk is only %d bytes", buffer_size, payload_size);
```

Comment:
So in `put_all_chunks` method we don't know if the user configured multipart mode or PutObject mode... hence, I think the log message here shouldn't use `upload_chunk_size` since that might be confusing. I prefer just "Pre-compression chunk size = %d... "

Reply:
Will do